pages/api/auction/createOrEditBid.ts

```diff
@@ -11,13 +11,18 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
   const { signature, order }: { signature: string; order: Order } = req.body
 
   const auction = (await getAuction()).data() as Auction
-  const oldBid = auction.bids[`${order.trader}-${order.nonce}`] as Bid
 
-  try {
-    const isOwner = verifyOrder(order, signature, oldBid?.bidder || order.trader)
-    if (!isOwner) return res.status(401).json({ message: 'Not owner' })
-  } catch (e) {
-    return res.status(401).json({ message: "Signature can't be verified" })
+  if (auction.bids !== undefined) {
+    const oldBid = auction.bids[`${order.trader}-${order.nonce}`] as Bid
+
+    try {
+      const isOwner = verifyOrder(order, signature, oldBid?.bidder || order.trader)
+      if (!isOwner) return res.status(401).json({ message: 'Not owner' })
+    } catch (e) {
+      return res.status(401).json({ message: "Signature can't be verified" })
+    }
+  } else {
+    auction.bids = {}
```

Comment:
I don't understand this case. Is there a situation where auction.bids is undefined?

Reply:
I was testing the submit_bid in the SDK, so I created quickly a new auction https://github.com/opynfinance/crab-v2-sdk-python/blob/a868e353bed834830a55d2bdf26833f9a53f81eb/scripts/crab_v2_sdk_script.py#L68 without passing the bids array, I think for now we can have this if/else statement to make it easier to test, until more improvement to the SDK tests.

Reply:
Also the API also should later have input validation checks, an auction without bids array shouldn't be created IMO if we are not counting a special case for it